pkg/controller/util/node/controller_utils.go

```diff
@@ -44,29 +42,22 @@ import (
 // DeletePods will delete all pods from master running on given node,
 // and return true if any pods were deleted, or were found pending
 // deletion.
-func DeletePods(kubeClient clientset.Interface, recorder record.EventRecorder, nodeName, nodeUID string, daemonStore appsv1listers.DaemonSetLister) (bool, error) {
+func DeletePods(kubeClient clientset.Interface, pods []v1.Pod, recorder record.EventRecorder, nodeName, nodeUID string, daemonStore appsv1listers.DaemonSetLister) (bool, error) {
```

Comment:
Can we split this PR even further to do a bit more of this refactoring? In particular, I think it should be possible to factor out to a separate PR:
- changes in the test (setting getPodsAssignedToNode function)
- changes to the interface to DeletePod (and for now just introduce a small helper function for listing pods from a node using the client)
- changes to the interface to MarkPodsReady

This would make: (1) the factored out changes no-op, so easy to review, (2) this PR much smaller (i.e. also much easier to review)

Reply:
> changes in the test (setting getPodsAssignedToNode function)

https://github.com/kubernetes/kubernetes/pull/83248

Reply:
> changes to the interface to DeletePod (and for now just introduce a small helper function for listing pods from a node using the client)

#83320

Reply:
Done.